src/gui_common/ModalManager.cs

```diff
@@ -15,7 +16,9 @@ public class ModalManager : NodeWithInput
     /// </summary>
     private readonly System.Collections.Generic.Dictionary<CustomWindow, Node> originalParents = new();
 
-    private readonly Stack<CustomWindow> modalStack = new();
+    private readonly Queue<CustomWindow> demotedModals = new();
+
+    private Stack<CustomWindow> modalStack = new();
```

Comment:
Can this be readonly as well?

Reply:
No, is used like this just few lines down
https://github.com/Revolutionary-Games/Thrive/blob/4cb9670399c902dd7ebea2feb04bbbe543ef94a0/src/gui_common/ModalManager.cs#L212-L221

Reply:
Well if that is a common problem, I think maybe a Deque should be used then instead to allow the expected error condition to be handled correctly?

Reply:
The situation can occur somewhere in the middle of the stack not just both ends. Not only that, we'd have to reimplement stack behavior

> Well if that is a common problem

I wouldn't say it's that common, and one line of LINQ for it shouldn't be much of a performance sap.

Reply:
Conceptually if the thing doesn't act like a stack or queue, then those data structures shouldn't be used. A deque sounds exactly what is needed here (we already have a Deque library), it supports push and pop to both front and back, while also supporting access to removing middle elements.